core/include/usModuleInfo.h

```diff
@@ -48,6 +49,7 @@ struct US_Core_EXPORT ModuleInfo
   std::string name;
   std::string location;
   std::string autoLoadDir;
+  std::map<std::string, std::string> properties;
```

Comment:
I agree. The core context is an appropriate place given that the Framework launching properties need to be accessible through the BundleContext's "getProperty()" function (per OSGi spec, section 4.5.3 - Environment Properties).

It does look like the removal of ModuleInfo will simplify the code. I would like to make this change in a different feature branch since it is a significant change. Making ModuleRegistry private to the framework can then be rolled into this separate feature branch as well.